src/Microsoft.ApplicationInsights/Managed/Shared/Extensibility/Implementation/Tracing/HeartbeatDefaultPayload.cs

```diff
@@ -0,0 +1,129 @@
+                        });
+                        break;
+                    case "baseSdkTargetFramework":
+                        payload.Add(fieldName, new HeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = this.GetTargetFrameworkVer()
+                        });
+                        break;
+                    case UpdatedFieldsPropertyKey:
+                        var updatedFieldItem = new HeartbeatPropertyPayload()
+                        {
+                            IsHealthy = true,
+                            PayloadValue = string.Empty
+                        };
+                        updatedFieldItem.IsUpdated = false; // always set this to false
+                        payload.Add(fieldName, updatedFieldItem);
+                        break;
+                    default:
+                        payload.Add(fieldName, new HeartbeatPropertyPayload()
+                        {
+                            IsHealthy = false,
+                            PayloadValue = "UNDEFINED"
+                        });
+                        break;
+                }
+            }
+
+            return payload;
+        }
+
+        private void SetEnabledProperties(IEnumerable<string> disabledFields)
+        {
+            if (disabledFields == null || disabledFields.Count() <= 0)
+            {
+                this.enabledProperties = DefaultFields.ToList();
+            }
+            else
+            {
+                this.enabledProperties = new List<string>();
+                foreach (string fieldName in DefaultFields)
+                {
+                    if (!disabledFields.Contains(fieldName, StringComparer.OrdinalIgnoreCase))
+                    {
+                        this.enabledProperties.Add(fieldName);
+                    }
+                }
+            }
+        }
+
+        private string GetTargetFrameworkVer()
+        {
+#if NET45
+            return "net45";
+#elif NET46
+            return "net46";
+#elif NETSTANDARD1_3
+            return "netstandard1.3";
+#else
+#error Unrecognized framework
+            return "undefined";
+#endif
+        }
+
+        private string GetRuntimeFrameworkVer()
+        {
+            // taken from https://github.com/Azure/azure-sdk-for-net/blob/f097add680f37908995fa2fbf6b7c73f11652ec7/src/SdkCommon/ClientRuntime/ClientRuntime/ServiceClient.cs#L214
+            try
+            {
+                Assembly assembly = typeof(Object).GetTypeInfo().Assembly;
+
+                AssemblyFileVersionAttribute objectAssemblyFileVer =
+                            assembly.GetCustomAttributes(typeof(AssemblyFileVersionAttribute))
+                                    .Cast<AssemblyFileVersionAttribute>()
+                                    .FirstOrDefault();
+
+                return objectAssemblyFileVer != null ? objectAssemblyFileVer.Version : "undefined";
```

Comment:
should we just cache the result? This is not going to change during application's lifetime right?

Reply:
This class is only called once on HeartbeatProvider.Initialize, the values are cached there and sent out with each heartbeat pulse.